src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -93,9 +94,14 @@ void AddRedirections(Dictionary<string, SourceInfo<string>> items, bool redirect
                                 errors.Add(Errors.RedirectionUrlInvalid(redirectUrl));
                                 continue;
                         }
+
+                        if (redirectUrl.Value.EndsWith("index"))
+                        {
+                            mutableRedirectUrl = redirectUrl.Value.Substring(0, redirectUrl.Value.LastIndexOf("index"));
```

Comment:
avoid scan in `LastIndexOf`:
```suggestion
                            mutableRedirectUrl = redirectUrl.Value.Substring(0, redirectUrl.Value.Length - "index".Length);
```